### PR TITLE
goat: init at unstable-2022-08-15

### PR DESCRIPTION
**Pull request by katexochen**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/go/goat/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
+,
+}:
+buildGoModule rec {
+  pname = "goat";
+  # Upstream currently isn't doing tags/releases, so using the Go pseudo-versioning instead.
+  # See https://pkg.go.dev/github.com/blampe/goat?tab=versions.
+  version = "v0.0.0-20220815015552-07bb911fe310";
```
Comment (Gerg-L) [inline pkgs/by-name/go/goat/package.nix:10]: Please use the standard versioning scheme 
`unstable-2022-08-15`

File: pkgs/by-name/go/goat/package.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchFromGitHub {
+    owner = "blampe";
+    repo = pname;
+    rev = "07bb911fe310";
```
Comment (Gerg-L) [inline pkgs/by-name/go/goat/package.nix:15]: Use the full rev `07bb911fe3106cc3c1d1097318a9fffe816b59fe`

File: pkgs/by-name/go/goat/package.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
+,
```
Comment (Gerg-L) [inline pkgs/by-name/go/goat/package.nix:4]:
```suggestion
, buildGoModule
, fetchFromGitHub
```

File: pkgs/by-name/go/goat/package.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}:
+buildGoModule rec {
```
Comment (Gerg-L) [inline pkgs/by-name/go/goat/package.nix:6]:
```suggestion
buildGoModule {
```

File: pkgs/by-name/go/goat/package.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "goat";
+  # Upstream currently isn't doing tags/releases, so using the Go pseudo-versioning instead.
+  # See https://pkg.go.dev/github.com/blampe/goat?tab=versions.
+  version = "v0.0.0-20220815015552-07bb911fe310";
+
+  src = fetchFromGitHub {
+    owner = "blampe";
+    repo = pname;
```
Comment (Gerg-L) [inline pkgs/by-name/go/goat/package.nix:14]:
```suggestion
    repo = "goat";
```

**Review by Gerg-L — CHANGES_REQUESTED** [review]
A few nitpicks, biggest thing is the `version`

File: pkgs/by-name/go/goat/package.nix (hunk 3 above)
Comment (katexochen) [reply pkgs/by-name/go/goat/package.nix]: Arrg, that bites me every time. :laughing: Wished the formatter would catch that...

File: pkgs/by-name/go/goat/package.nix (hunk 1 above)
Comment (katexochen) [reply pkgs/by-name/go/goat/package.nix]: I this documented somewhere? I'm always a bit lost where to find these standards.

Thanks for reviewing!

File: pkgs/by-name/go/goat/package.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
```
Comment (Gerg-L) [inline pkgs/by-name/go/goat/package.nix:3]: Switch the order of these
It goes lib, stdenv/your builder then fetchers, then everything else ASCII sorted

**Review by Gerg-L — APPROVED** [review]
Missed one change, otherwise looks good to me

Result of `nixpkgs-review pr 257103` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>goat</li>
  </ul>
</details>

File: pkgs/by-name/go/goat/package.nix (hunk 6 above)
Comment (katexochen) [reply pkgs/by-name/go/goat/package.nix]: Again, is this documented anywhere?

**Review by Gerg-L — APPROVED** [review]

File: pkgs/by-name/go/goat/package.nix (hunk 6 above)
Comment (wegank) [reply pkgs/by-name/go/goat/package.nix]: The order is described in the last line of https://nixos.org/manual/nixpkgs/stable/#sec-syntax. The change seems to be correct somehow.

File: pkgs/by-name/go/goat/package.nix (hunk 1 above)
Comment (wegank) [reply pkgs/by-name/go/goat/package.nix]: https://nixos.org/manual/nixpkgs/stable/#sec-package-naming

**Pull request merged**